plugins/auto-sizes/tests/improve-sizes-test.php

```diff
@@ -24,85 +24,198 @@ public static function set_up_before_class(): void {
 		switch_theme( 'twentytwentyfour' );
 
 		self::$image_id = self::factory()->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/data/images/leaves.jpg' );
+
+		// Disable lazy loading attribute.
+		add_filter( 'wp_img_tag_add_loading_attr', '__return_false' );
```

Comment:
This is a risky filter to disable while running these tests, because we'll be missing scenarios where the loading attribute changes the way the markup gets created. I assume you did this to avoid having auto sizes added while running these tests? If so, instead of removing the `wp_img_tag_add_loading_attr` filter, I would suggest running these without the auto-sizes filter, which seems to be what is causing potential failures here.
```suggestion
		// Disable auto sizes in these tests.
		remove_filter( 'wp_content_img_tag', 'auto_sizes_update_content_img_tag' );
```

However, since the hooks global won't get reset after this class runs, you'll also need to add this hook back after the class runs, or else it will affect any tests that run afterwards. Ex:

```php
	public static function tear_down_after_class(): void {
		parent::tear_down_after_class();

		add_filter( 'wp_content_img_tag', 'auto_sizes_update_content_img_tag' );
	}
```

If there are specific tests where you need to disable lazy-loading, it would be better to modify hooks directly in that test, because then they will automatically get reset during the `WP_UnitTestCase_Base::tear_down()` method that runs after each test.

Reply:
Thanks @joemcgill for suggestion. This will not working as the filter is re-added through `the_content` filter so it show error in tests.

```
1) Tests_Improve_Sizes::test_image_block_with_full_alignment with data set "Return wideSize 1280px instead of medium size 300px" ('medium')
Failed asserting that '<figure class="wp-block-image size-medium"><img loading="lazy" decoding="async" width="300" height="200"   src="http://localhost:8889/wp-content/uploads/2024/06/leaves-283-300x200.jpg" alt="" class="wp-image-5" srcset="http://localhost:8889/wp-content/uploads/2024/06/leaves-283-300x200.jpg 300w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283-1024x683.jpg 1024w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283-768x512.jpg 768w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283.jpg 1080w" sizes="auto, 100vw" /></figure>' contains "sizes="100vw" ".

/var/www/html/wp-content/plugins/performance/plugins/auto-sizes/tests/test-improve-sizes.php:53

2) Tests_Improve_Sizes::test_image_block_with_full_alignment with data set "Return wideSize 1280px instead of large size 1024px" ('large')
Failed asserting that '<figure class="wp-block-image size-large"><img loading="lazy" decoding="async" width="1024" height="683"   src="http://localhost:8889/wp-content/uploads/2024/06/leaves-283-1024x683.jpg" alt="" class="wp-image-5" srcset="http://localhost:8889/wp-content/uploads/2024/06/leaves-283-1024x683.jpg 1024w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283-300x200.jpg 300w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283-768x512.jpg 768w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283.jpg 1080w" sizes="auto, 100vw" /></figure>' contains "sizes="100vw" ".

/var/www/html/wp-content/plugins/performance/plugins/auto-sizes/tests/test-improve-sizes.php:53

3) Tests_Improve_Sizes::test_image_block_with_full_alignment with data set "Return wideSize 1280px instead of full size 1080px" ('full')
Failed asserting that '<figure class="wp-block-image size-full"><img loading="lazy" decoding="async" width="1080" height="720"   src="http://localhost:8889/wp-content/uploads/2024/06/leaves-283.jpg" alt="" class="wp-image-5" srcset="http://localhost:8889/wp-content/uploads/2024/06/leaves-283.jpg 1080w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283-300x200.jpg 300w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283-1024x683.jpg 1024w, http://localhost:8889/wp-content/uploads/2024/06/leaves-283-768x512.jpg 768w" sizes="auto, 100vw" /></figure>' contains "sizes="100vw" ".

/var/www/html/wp-content/plugins/performance/plugins/auto-sizes/tests/test-improve-sizes.php:53

```